app/models/organization.rb

```diff
@@ -30,13 +33,17 @@ def self.not_deleted
   end
 
   def soft_delete
-    return false if open_orders.present?
+    ensure_no_open_orders
 
-    transaction do
-      organization_users.map(&:destroy!) if organization_users.present?
+    begin
+      transaction do
+        organization_users.map(&:destroy!) if organization_users.present?
 
-      self.deleted_at = Time.zone.now
-      save
+        self.deleted_at = Time.zone.now
+        save!
+      end
+    rescue
+      raise DeletionError
```

Comment:
Why rescue and re-raise as DeletionError? You will lose the original stack trace and message, or is that what you intended?